src/parser/mod.rs

```diff
@@ -7646,6 +7646,39 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_index_exprs(&mut self) -> Result<Vec<IndexExpr>, ParserError> {
+        self.parse_parenthesized(|p| p.parse_comma_separated(Parser::parse_index_expr))
+    }
+
+    pub fn parse_index_expr(&mut self) -> Result<IndexExpr, ParserError> {
+        let expr = self.parse_expr()?;
+        let collation = if self.parse_keyword(Keyword::COLLATE) {
+            Some(self.parse_object_name(false)?)
+        } else {
+            None
+        };
+
+        let (operator_class, order_options) = if self.peek_keyword(Keyword::ASC)
+            || self.peek_keyword(Keyword::DESC)
+            || self.peek_keyword(Keyword::NULLS)
```

Comment:
can we introduce a test case that covers the `ASC/DESC` and `nulls_first` syntax?